### DISCUSSION TITLE
Security State?

### DISCUSSION THREAD

**Discussion by kitty99hub** [jmcollin78/versatile_thermostat#238] [Q&A]
Hi,
Since 2 days, I have had problems with my versatile thermostat.
I used it to control an electrical radiator, and it worked fine for about 2 weeks.
If I turn the thermostat to "heat", after a short time (one or two minutes), it turns automatically off.
Now I have seen that it has an entity called "security state" and that this was tripped to "unsafe"....
Unfortunately, I could not find any information about this entity and the state "unsafe".
Any idea what will cause such an unsafe state?
Many thanks
kind regards

**Comment by paulashby1966** [comment]
I have a similar issue to yours that has started in the last week or two.
I have IR panels controlled by zigbee sockets and motion detectors. They are often now in security when I arrive home and the rooms are way below the ECO setpoint. I wanted to ask why this happens and if anyone has some ideal advanced settings to stop it happening. I changed some this morning but when I got home most had gone to security again.

**Comment by jmcollin78** [comment]
Security state are explained in the readme. Please have a look.

**Reply by kitty99hub** [reply]
Hi @jmcollin78 
thanks for your reply.
I already had read it, but unfortunately I don't really understand it and couldn't find out what the exact reason for my failures is.
I don't want to start messing around with all the parameters to try it by the try and error method ;) 
I'm missing either a good practical example or a more detailed description of this function....
Attached is a screenshot of the switching process, maybe anybody can read something suspicious out of it...?
Thanks,
regards
![image](https://github.com/jmcollin78/versatile_thermostat/assets/115743879/ad2aec5b-7cc2-4eb5-848d-0c858e146175)

**Comment by jmcollin78** [comment]
Security mode is when you temperature sensor stops giving the temperature. This can be very dangerous, if the heater is then stuck at 100% heating.

So there is delay configurable in the "Advanced configuration" page (the last page).
If I look at your temperature curve, it seems that temperature have not been send during more than one hour, just before the security state is on:
![Capture d’écran 2023-12-01 à 17 25 39](https://github.com/jmcollin78/versatile_thermostat/assets/1712374/264c1042-3cc8-43f4-864e-3736b89b95bc)

If you consider it is normal, you "just" have to change the configuration parameter describe in readme:
```
The second delay (security_delay_min) is the maximal delay between two temperature measure before setting the preset to security and turning off the thermostat. If the temperature sensor is no more giving temperature measures, the thermostat and heater will turns off after this delay and the preset of the thermostat will be set to security. This is useful to avoid overheating is the battery of your temperature sensor is too low.
```

Beware this is also the case for the outdoor temperature. With the VTherm UI Card (https://github.com/jmcollin78/versatile-thermostat-ui-card) you know when this happens and why.

What seems not normal in that the security should stops when the temperature sensor sends the temperature. And it seems it is not stopping as expected. I will have a look.

**Reply by kitty99hub** [reply]
Hi and thanks for the infos.
In following screenshot I added the outside temperature and it seems for mee, that there was no signal outage for the in and outside temperatures.
![image](https://github.com/jmcollin78/versatile_thermostat/assets/115743879/0cda5225-93cc-4d6b-a8a2-c0a1cd0c1de0)

**Comment by jmcollin78** [comment]
I cannot reproduce your case. All works fine on my env.

**Reply by kitty99hub** [reply]
Hi @jmcollin78 
ok, I'll continue to observe it and if it happens again I'll see if I can provide further information.
Maybe an idea or tip on what I should take a look at, i.e. what information might be interesting for investigation?
thanks and have a nice weekend

**Reply by jmcollin78** [reply]
Yes please, if it is reproducing, can you provide all attributes of the VTherm ( Development Tools / State in text format and formatted with the <> button).
If possible I would like also some logs when the VTherm stays in Security mode but should not.

This would be very helpful. Thank you in advance

**Comment by FranMaverix** [comment]
Hi, I'm having that trouble in my two new created versatile thermostat (two electrical radiators)
Right now, I have the two in security mode. Both temp. sensors are available.

The climate.salita state is this: (it shows this warning in the card: internal temp: 20 min.)
----------------------------------
hvac_modes:
  - heat
  - "off"
min_temp: 10
max_temp: 28
target_temp_step: 0.1
preset_modes:
  - none
  - eco
  - comfort
  - boost
current_temperature: 21.6
temperature: 22.5
hvac_action: idle
preset_mode: security
is_on: true
hvac_mode: heat
type: null
eco_temp: 21
boost_temp: 24
comfort_temp: 22.5
eco_away_temp: 18
boost_away_temp: 20
comfort_away_temp: 19
power_temp: 20
target_temperature_step: 0.1
ext_current_temperature: 17
ac_mode: false
current_power: null
current_power_max: null
saved_preset_mode: comfort
saved_target_temp: 21
saved_hvac_mode: heat
window_state: "off"
motion_state: null
overpowering_state: null
presence_state: "on"
window_auto_state: "off"
window_bypass_state: false
security_delay_min: 60
security_min_on_percent: 0.5
security_default_on_percent: 0.1
last_temperature_datetime: "2023-12-06T18:24:07.235535+01:00"
last_ext_temperature_datetime: "2023-12-06T13:04:35.164367+01:00"
security_state: true
minimal_activation_delay_sec: 10
device_power: 2.4
mean_cycle_power: 0.24
total_energy: 5.25
last_update_datetime: "2023-12-06T18:24:25.144084+01:00"
timezone: Europe/Madrid
window_sensor_entity_id: null
window_delay_sec: 30
window_auto_open_threshold: 3
window_auto_close_threshold: 0
window_auto_max_duration: 30
motion_sensor_entity_id: null
presence_sensor_entity_id: input_boolean.presencia_salita
power_sensor_entity_id: null
max_power_sensor_entity_id: null
temperature_unit: °C
is_device_active: false
ema_temp: 21.76
is_over_switch: true
is_inversed: false
underlying_switch_0: switch.salita_3
underlying_switch_1: null
underlying_switch_2: null
underlying_switch_3: null
on_percent: 0.1
power_percent: 10
on_time_sec: 90
off_time_sec: 810
cycle_min: 15
function: tpi
tpi_coef_int: 0.6
tpi_coef_ext: 0.01
friendly_name: Salita
supported_features: 17


The other device only shows in the lovelace card an thermometer with exclamation without text.
state:
-------------------------
hvac_modes:
  - heat
  - "off"
min_temp: 7
max_temp: 35
target_temp_step: 0.1
preset_modes:
  - none
  - eco
  - comfort
  - boost
current_temperature: 18.5
temperature: 16.9
hvac_action: "off"
preset_mode: none
is_on: false
hvac_mode: "off"
type: null
eco_temp: 19
boost_temp: 22.5
comfort_temp: 21
eco_away_temp: 18
boost_away_temp: 20
comfort_away_temp: 19
power_temp: 18
target_temperature_step: 0.1
ext_current_temperature: 17
ac_mode: false
current_power: 0
current_power_max: null
saved_preset_mode: eco
saved_target_temp: 18
saved_hvac_mode: heat
window_state: "off"
motion_state: null
overpowering_state: null
presence_state: "on"
window_auto_state: "off"
window_bypass_state: false
security_delay_min: 60
security_min_on_percent: 0.5
security_default_on_percent: 0.1
last_temperature_datetime: "2023-12-06T18:43:28.346010+01:00"
last_ext_temperature_datetime: "2023-12-06T13:04:35.164367+01:00"
security_state: true
minimal_activation_delay_sec: 10
device_power: 1.5
mean_cycle_power: 0.15000000000000002
total_energy: 337.81
last_update_datetime: "2023-12-06T18:43:28.351103+01:00"
timezone: Europe/Madrid
window_sensor_entity_id: binary_sensor.ventana_cuarto
window_delay_sec: 30
window_auto_open_threshold: null
window_auto_close_threshold: null
window_auto_max_duration: null
motion_sensor_entity_id: null
presence_sensor_entity_id: input_boolean.presencia_cuarto
power_sensor_entity_id: sensor.regleta_mesilla
max_power_sensor_entity_id: null
temperature_unit: °C
is_device_active: false
ema_temp: 18.62
is_over_switch: true
is_inversed: false
underlying_switch_0: switch.regleta2_2
underlying_switch_1: null
underlying_switch_2: null
underlying_switch_3: null
on_percent: 0.1
power_percent: 10
on_time_sec: 60
off_time_sec: 540
cycle_min: 10
function: tpi
tpi_coef_int: 0.6
tpi_coef_ext: 0.01
friendly_name: Cuarto
supported_features: 17


I hope this help.

**Reply by jmcollin78** [reply]
Hello @FranMaverix ,

Your problem is the outdoor temperature sensor for which the last timestamp of measure is:
`last_ext_temperature_datetime: "2023-12-06T13:04:35.164367+01:00" `
and now is `last_update_datetime: "2023-12-06T18:43:28.351103+01:00"` 5 hours later.

It is not visible in the UI card due to a bug that I have just publish, a few minutes ago. You should install it and security mode will be much mode explicit.

You have to check your outdoor temp sensor.

This explains that the both VTherm have turn into security mode in the same time.

**Reply by FranMaverix** [reply]
I was using the AEMET (Spainish Meteorology Agency) Integration as external temperature sensor. I switched to ACCUWEATHER integration and for now, is running OK.
I think that the forecast AEMET API do not allow too many calls. 
Thanks for your support and your work!

**Reply by maia** [reply]
A small suggestion: use the [ha-average](https://github.com/Limych/ha-average) sensor and create an average of multiple sources, ideally one of them local and one of them internet-based. I'm using this in more and more places, as it's nearly impossible to be unavailable.

This is what I use in `configuration.yaml` (replace the sensors with their real names):
```
  - platform: average
    name: "Outdoor temperature"
    # this will create the entity_id: sensor.outdoor_temperature
    unique_id: "outdoor_temperature"
    entities:
      - sensor.locally_measured_temperature
      - sensor.openweathermap_temperature
      - sensor.amet_temperature
      - sensor.accuweather_temperature
    duration:
      minutes: 10
    precision: 2
```

**Comment by jmcollin78** [comment]
@kitty99hub 

Maybe you have the same problem: the outdoor temperature sensor. You send me curve about the internal temp sensor but security mode can also turn on if outdoor temp is too old.

**Reply by kitty99hub** [reply]
Hi @jmcollin78 
many thanks for your answers and the tip about the outside temperature.
But I was able to rule that out, as far as I can tell from the graphs...
I'll now wait until the status appears again and then read out the attributes and post them here.
Here a screen shot from one week with a lot of outages, where you can see the room and the outside temperature, aswell the security state.
![image](https://github.com/jmcollin78/versatile_thermostat/assets/115743879/3c65e389-8d1d-48cd-9ba0-deb1436e0cd7)

**Reply by jmcollin78** [reply]
Hello, where is this curve in your graph ?

![Capture d’écran 2023-12-08 à 17 08 30](https://github.com/jmcollin78/versatile_thermostat/assets/1712374/b81a3dce-e7df-42b6-8499-a5a84ecacb7e)

I don't see it or I need some holidays maybe ...

You can also install the last VTherm UI Card. It says now clearly when it is in security which temperature is missing with a window like this:

![Capture d’écran 2023-12-08 à 17 23 42](https://github.com/jmcollin78/versatile_thermostat/assets/1712374/ae3de79a-1178-4185-8ce8-bda70c636cc1)

**Reply by kitty99hub** [reply]
Hi @jmcollin78 
I had to look again myself... :)
There are two overlapping curves !
The current room temperature is actually shown twice on this combined graph.
Once the entity that comes from the Aqara temperature sensor itself: sensor.bedroom_multisensor_aqaralumiweather_temperature (shown as orange line) and once supplied by the Versatile Thermostat entity: climate.bedroom_heater (shown as dark blue line).
![image](https://github.com/jmcollin78/versatile_thermostat/assets/115743879/e2b41379-14b0-42d0-bfcf-bff0eb3469d7)

In the two following graphics, where only these entities are visible, you can see that they represent the same data and therefore overlap.
![image](https://github.com/jmcollin78/versatile_thermostat/assets/115743879/3f5ad8a8-eb82-435a-b597-701bfc1467cd)

![image](https://github.com/jmcollin78/versatile_thermostat/assets/115743879/ffbfaff2-33a4-4a91-b037-a2dd906feaec)

Sorry for this misleading.
Yes, I could try out the VTherm UI card for testing purposes.
I didn't want to install it until now because I don't actually intend to use it on my system.
Based on the history, I can't determine that the security mode is caused by failures of the temperature sensors...
Can this also be triggered by other causes/conditions?
thanks

**Reply by jmcollin78** [reply]
> Can this also be triggered by other causes/conditions?

No. It is the only condition.

**Reply by jmcollin78** [reply]
Once again, if you have some Security mode problem, I need the VTherm attributes when it is in security. They are in Development Tools / States. Post me that in yaml formatted with the <> button of the editor.

Look at the attributes named: 
```
last_temperature_datetime: "2023-12-06T18:43:28.346010+01:00"
last_ext_temperature_datetime: "2023-12-06T13:04:35.164367+01:00"
security_state: true
last_update_datetime: "2023-12-06T18:43:28.351103+01:00"
```

You should a gap between last_temperature_datetime or last_ext_temperature_datetime and last_update_datetime (the current date). Then you will be able to determine which temperature sensor is in trouble.

**Reply by kitty99hub** [reply]
Hi @jmcollin78 
> Once again, if you have some Security mode problem, I need the VTherm attributes when it is in security. 

Yes I know, I am still waiting to capture the next outage.
It's now running again for a while without any failures.
As soon as the security mode appears again and I can readout the attributes, I will post them here.
thanks and have a nice Sunday

**Reply by jmcollin78** [reply]
Ah ok. I think you ran into the security mode one again.

You have all the operations you can follow to diagnose the Security mode here : https://github.com/jmcollin78/versatile_thermostat/blob/main/README.md#why-does-my-versatile-thermostat-go-into-safety
